### DISCUSSION TITLE
checkbox implementation (example-sim exercises)

### DISCUSSION THREAD

**Discussion by dave-english** [scenerystack/community#51] [Q&A] [answered]
I'm trying to add a checkbox to the example-sim (as an exercise)
I create an instance of the checkbox class, imported from:` import Checkbox from '../../../../sun/js/Checkbox.js';`
One of the parameters for the checkbox seems to be an object of the PhetioProperty class. However, I'm having some problems invoking instances of this class.

When I begin to type "Import PhetioProperty..."  (in VSCode), The file location does not autocomplete, as it does for nearly every other "Import" command I've used.
On the other hand, if I manually enter the file location, as 
`import PhetioProperty from '../../../../axon/js/PhetioProperty.js';
`
then later generate an instance of the PhetioProperty class within a constructor definition in the same file, as
`this.isVisibleProperty = new PhetioProperty(true);` 
 the code will launch, but the browser throws an error:

FerroParticle.js:9 Uncaught SyntaxError: The requested module '../../../../axon/js/PhetioProperty.js' does not provide an export named 'default' (at FerroParticle.js:9:8)

What's happening here?  Why isn't PhetioProperty recognized by VSCode? Or am I taking entirely the wrong approach to setting up my checkbox and its parameters?

**Comment by samreid** [comment]
From https://github.com/phetsims/axon/blob/96745eaf3f43485e19f32ca941a0f5add2ef4911/js/PhetioProperty.ts#L13 I can see that `PhetioProperty` is a type alias and therefore cannot be instantiated. I'm not sure why VSCode isn't autocompleting it (perhaps it deals with types differently?). But perhaps `new BooleanProperty(true)` would work in this case.

**Reply by dave-english** [reply, marked as answer]
Sam Reid, Thank you.
Hours after posting my question (but before I saw your answer) I solved problem 3 in the exercises (hiding and unhiding a ball from the screen with a checkbox)
I was bedeviled by three problems, which I solved in the following order:
1) I stopped trying to create a PhetioProperty as the parameter of my checkbox, and changed it to a BooleanProperty instead
2) The container in which I was trying to display my checkbox was simply too small, and kept getting snagged by some internal self pruner function. 
3) I was under the misconception that the boolean parameter in the checkbox was associated with a boolean variable in the model. When I replaced that parameter with  a boolean from the view screen node instead (and also formatted the node listener correctly) the graphic turned on and off with the checkbox, just as I wanted.

To format the node listener correctly, I was helped greatly by the "make-a-ten" example.

```
    this.hideSumProperty = new BooleanProperty(false);
    this.hideSumProperty.link( hideSum => {
          this.visible = !hideSum;    
        } );  
```
